File: 240507/정수 명령 처리/process-numeric-commands.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        Stack stack = new Stack<Integer>();
        for(int i = 0; i < n; i++) {
            String cmd = sc.next();

            switch(cmd) {
                case "push": 
                    int item = sc.nextInt();
                    stack.push(item);
                    break;
                
                case "pop":
                    System.out.println(stack.pop());
                    break;
                
                case "size":
                    System.out.println(stack.size());
                    break;
                
                case "empty":
                    if(stack.isEmpty()){
                        System.out.println(1);
                    }else {
                        System.out.println(0);
                    }
                    break;
                case "top" :
                    System.out.println(stack.peek());
                    break;
            }
        }
    }
}

File: 240507/정수 명령 처리/process-numeric-commands.kt
import java.util.*

fun main(){
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val stack = Stack<Int>()

    for(i in 0 until n) {
        val cmd = sc.next().toString()
        when(cmd) {
            "push" -> {
                val item = sc.nextInt()
                stack.push(item)
            }
            "pop" -> {
                val item = stack.pop()
                println(item)
            }
            "size" -> {
                println(stack.size)
            }
            "empty" -> {
                if(stack.isEmpty()) println(1) else println(0)
            }
            "top" -> {
                println(stack.peek())
            }
        }
    }
}